Products/PerFactErrors/errors.py

```diff
@@ -50,7 +51,53 @@ def afterfail_error_message(event):
         if body is not None:
             req.response.setBody(body)
     except Exception:
+        logger.warn(traceback.format_exc())
         logger.exception('Error while rendering error message')
         transaction.abort()
     else:  # no exception
         transaction.commit()
+
+
+class PerFactException(Exception):
+    ''' Special Exception Class
```

Comment:
A sentence here would be nice explaining the purpose of the class. Most of it can be deduced from the docstring of `__init__`, but anyhow.